csvcubed/csvcubed/models/cube/qb/components/codelist.py

```diff
@@ -159,8 +157,28 @@ def get_permitted_rdf_fragment_hints(self) -> Set[RdfSerialisationHint]:
     def get_default_node_serialisation_hint(self) -> RdfSerialisationHint:
         return RdfSerialisationHint.ConceptScheme
 
-    def validate_data(self, data: PandasDataTypes) -> List[ValidationError]:
-        return []  # TODO: implement this.
+    def validate_data(self, data: PandasDataTypes) -> list[Optional[ValidationError]]:
+        """
+        Validate the data held in the codelists, assuming case insensitivity
+        """
+
+        errors = list()
+
+        if data is not None:
+            unique_keys = dict(list)
```

Comment:
`dict(list)` what is this?

Reply:
It's a dictionary where the values must be lists.